realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1553,6 +1553,10 @@ private void emitCreateOrUpdateUsingJsonObject(JavaWriter writer) throws IOExcep
         writer.emitEmptyLine();
     }
 
+    // FIXME: Since we need to check the PK in stream before create an object, this is now using copyToRealm instead of
```

Comment:
It looks like it only effect the `create` part, not the `update` part right?
I see 3 choices:

1) Accept the current solution and document the breaking change 
2) Rewrite `createFromJsonStream` to create JSONObject instead and return the result from `createObjectFromJson(jsonObj)`
3) Add support for #777 

For expendiency, I would probably just go with 1) now make an issue for looking into 2/3 in th next week or two, because we need a better solution before releasing anything publically.

If 3) is indeed fairly easy to do that would be awesome, if not I think we should do 2.